File: 141. Linked List Cycle/Unordered_set/LinkedListCycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    if(head === null)
        return false;

    const visited = new Set();
    let temp = head;

    while (temp != null){
        if (visited.has(temp))
            return true;

        visited.add(temp);
        temp = temp.next
    }
    return false;
};

File: 141. Linked List Cycle/Unordered_set/LinkedListCycle.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public bool HasCycle(ListNode head) {
        if(head == null){
            return false;
        }

        HashSet<ListNode> visited = new HashSet<ListNode>();
        ListNode temp = head;

        while(temp != null){
            if(visited.Contains(temp)){
                return true;
            }

            visited.Add(temp);
            temp = temp.next;
        }

        return false;
    }
}